stories/utils/generate-widget-stories.js

```diff
@@ -81,27 +82,40 @@ export function generateReportBasedWidgetStories( {
 	group,
 	data,
 	options,
+	variantCallbacks = {},
```

Comment:
I think we could simplify this function. The `variantCallbacks` introduces some complexity and there's a ton of similar code in the Storybook definition above that could be omitted. How about this alternative:
* Both `args.data` and `args.options` can also be arrays of multiple instances of data / options.
* If `args.options` is an array, `args.data` must be an array (of arrays) too. Their lengths must match each other.
* If any of this is not the case (invalid function call), throw an error.
* Otherwise, if they are both arrays, iterate through them and call `receiveGetReport`, `receiveError` for `getReport` etc. accordingly.

This way it's more automatic and avoids the need for writing all the callback functions above.

Reply:
@felixarntz this approach will work as long as we are always expecting to dispatch `dispatch( datastore ).receiveGetReport( data, { options } );` Unfortunately, this isn't always the case.

For example the Goals widget needs to dispatch `dispatch( STORE_NAME ).receiveGetGoals( goals );` as well. 

Perhaps, we can extend this approach by only using `variantCallback` to hold additional functions to run after we have iterated over the data/options array?

Reply:
@felixarntz I've implemented this in my commit here. This approach seems to work the best as we only need to populate the `additionalVariantCallbacks` objects when we deviate only needed to run `receiveGetReport`.